src/api/users.js

```diff
@@ -0,0 +1,393 @@
+import Q from "q";
+import logger from "winston";
+import moment from "moment";
+import randtoken from "rand-token";
+import atna from "atna-audit";
+import os from "os";
+import { User } from "../model/users";
+import * as authorisation from "./authorisation";
+import * as contact from "../contact";
+import { config } from "../config";
+import * as utils from "../utils";
+import * as auditing from "../auditing";
+
+config.newUserExpiry = config.get("newUserExpiry");
+config.userPasswordResetExpiry = config.get("userPasswordResetExpiry");
+config.alerts = config.get("alerts");
+
+const himSourceID = config.get("auditing").auditEvents.auditSourceID;
+
+/*
+ * Get authentication details
+ */
+export function* authenticate(email) {
+	email = unescape(email);
+
+	try {
+		const user = yield User.findOne({ email }).exec();
+
+		if (!user) {
+			utils.logAndSetResponse(this, 404, `Could not find user by email ${email}`, "info");
+			// Audit unknown user requested
+			let audit = atna.userLoginAudit(atna.OUTCOME_SERIOUS_FAILURE, himSourceID, os.hostname(), email);
+			audit = atna.wrapInSyslog(audit);
+			return auditing.sendAuditEvent(audit, () => logger.debug("Processed internal audit"));
+		} else {
+			return this.body = {
+				salt: user.passwordSalt,
+				ts: new Date()
+			};
+		}
+	} catch (e) {
+		return utils.logAndSetResponse(this, 500, `Error during authentication ${e}`, "error");
+	}
+}
+
+
+// ################################
+/* Reset password Functions */
+// ################################
+
+
+const passwordResetPlainMessageTemplate = (firstname, setPasswordLink) => `\
+<---------- Existing User - Reset Password ---------->
+Hi ${firstname},
+
+A request has been made to reset your password on the OpenHIM instance running on ${config.alerts.himInstance}
+Follow the below link to reset your password and log into OpenHIM Console
+${setPasswordLink}
+<---------- Existing User - Reset Password ---------->\
+`;
+
+const passwordResetHtmlMessageTemplate = (firstname, setPasswordLink) => `\
+<h1>Reset OpenHIM Password</h1>
+<p>Hi ${firstname},<br/><br/>A request has been made to reset your password on the OpenHIM instance running on ${config.alerts.himInstance}</p>
+<p>Follow the below link to set your password and log into OpenHIM Console</p>
+<p>${setPasswordLink}</p>\
+`;
+
+export function generateRandomToken() {
+	return randtoken.generate(32);
+}
+
+/*
+ * update user token/expiry and send new password email
+ */
+export function* userPasswordResetRequest(email) {
+	email = unescape(email);
+
+	if (email === "root@openhim.org") {
+		this.body = "Cannot request password reset for 'root@openhim.org'";
+		this.status = 403;
+		return;
+	}
+
+	// Generate the new user token here
+	// set expiry date = true
+
+	const token = exports.generateRandomToken();
+	const { duration } = config.userPasswordResetExpiry;
+	const { durationType } = config.userPasswordResetExpiry;
```

Comment:
Consolidated the destructuring. I guess the logic the conversion used is if you assigned a variable that is named the same as a member of an object then destructuring makes more sense. You are stating the member once vs once for the variable declaration and once again for access.